client/checkout/classic/new-upe.js

```diff
@@ -0,0 +1,242 @@
+	initAppearance();
+	initStripeElements();
+
+	function initAppearance() {
+		if ( ! appearance ) {
+			appearance = getAppearance();
+			api.saveUPEAppearance( appearance );
+		}
+	}
+
+	function initStripeElements() {
+		for ( const paymentMethodType in paymentMethods ) {
+			gatewayUPEComponents[ paymentMethodType ] = {
+				elements: null,
+				upeElement: null,
+			};
+		}
+	}
+
+	/**
+	 * This is the entry-point of this file, which is responsible for mounting the UPE elements.
+	 * It is called when the page is loaded.
+	 *
+	 */
+	$( document.body ).on( 'updated_checkout', () => {
+		$( '.wcpay-upe-element' )
+			.toArray()
+			.forEach( ( domElement ) => {
+				const paymentMethodType = $( domElement ).data(
+					'payment-method-type'
+				);
+				const upeElement =
+					gatewayUPEComponents[ paymentMethodType ].upeElement;
+				if ( upeElement ) {
+					upeElement.mount( domElement );
+				} else {
+					createAndMountPaymentElement(
+						paymentMethodType,
+						domElement
+					);
+				}
+			} );
+	} );
+
+	/**
+	 * Block UI to indicate processing and avoid duplicate submission.
+	 *
+	 * @param {Object} $form The jQuery object for the form.
+	 */
+	const blockUI = ( $form ) => {
+		$form.addClass( 'processing' ).block( {
+			message: null,
+			overlayCSS: {
+				background: '#fff',
+				opacity: 0.6,
+			},
+		} );
+	};
+
+	/**
+	 * Creates the UPE element and mounts it to the DOM element.
+	 * No Stripe PaymentIntent is created at this point.
+	 *
+	 * @param {string} paymentMethodType The type of payment method, e.g. 'card', 'giropay', etc.
+	 * @param {string} domElement The selector of the DOM element of particular payment method to mount the UPE element to.
+	 */
+	async function createAndMountPaymentElement(
+		paymentMethodType,
+		domElement
+	) {
+		const amount = Number( getUPEConfig( 'cartTotal' ) );
+		const options = {
+			mode: 1 > amount ? 'setup' : 'payment',
+			currency: getUPEConfig( 'currency' ).toLowerCase(),
+			amount: Number( getUPEConfig( 'cartTotal' ) ),
+			paymentMethodCreation: 'manual',
+			paymentMethodTypes: [ paymentMethodType ],
+			appearance: appearance,
+		};
+
+		if ( ! fingerprint ) {
+			try {
+				const { visitorId } = await getFingerprint();
+				fingerprint = visitorId;
+			} catch ( error ) {
+				// Do not mount element if fingerprinting is not available
+				showErrorCheckout( error.message );
+
+				return;
+			}
+		}
+
+		const elements = api.getStripe().elements( options );
+		gatewayUPEComponents[ paymentMethodType ].elements = elements;
+
+		const upeElement = elements.create( 'payment', {
+			...getUpeSettings(),
+			wallets: {
+				applePay: 'never',
+				googlePay: 'never',
+			},
+		} );
+
+		upeElement.mount( domElement );
+		gatewayUPEComponents[ paymentMethodType ].upeElement = upeElement;
+	}
+
+	let paymentMethodGenerated;
+
+	const handleCheckout = ( $form, paymentMethodType ) => {
+		if ( paymentMethodGenerated ) {
+			paymentMethodGenerated = null;
```

Comment:
```suggestion
			paymentMethodGenerated = false;
```
(I know--this is borrowed from the legacy card implementation, but...) AFAICT, the only values possible for `paymentMethodGenerated` are `true` or `null`. Why not `true` or `false` instead?

Reply:
Agree with you, updated in 48609a2b00e9a82a175018f2b16c0be667aa7a28. Thanks!